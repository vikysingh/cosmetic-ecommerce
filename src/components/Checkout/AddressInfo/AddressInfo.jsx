import React from "react"

import { Typography, Input, Select, Breadcrumb } from "antd"
import { Link } from "react-router-dom"
import checkouts from "../../../constants/checkouts.json"
import routes from "../../../constants/routes.json"
import { ThemeButton } from "../../index"

import styles from "./AddressInfo.module.css"

const { Title } = Typography
const { Option } = Select
const { Item } = Breadcrumb 

export default function AddressInfo() {
    return <div className={styles.CHECKOUT__LEFT_COL} >
        <Breadcrumb className={styles.CHECKOUT__LEFT_COL__breadcrumb} >
            <Item>
                <Link to={routes.home}> Home </Link>
            </Item>
            <Item>
                <Link to={routes.cart}> Cart </Link>
            </Item>
            <Item>
                Address
            </Item>
        </Breadcrumb>
        <Title level={3}> Contact information </Title>
        <Input type="email" placeholder="Email" />
        <form>
            <Title level={4}> Shipping address </Title>
            <Input type="text" placeholder="First Name" />
            <Input type="text" placeholder="Last Name" />
            <Input type="text" placeholder="Address" />
            <Input type="text" placeholder="Apartment, suite etc.(optional)" />
            <Input type="text" placeholder="City" />
            <Select showSearch className={styles.CHECKOUT__LEFT_COL__select}
            placeholder="Country"
            optionFilterProp="children"
            filterOption={(input, option) =>
            option.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
            }
            >
                { checkouts.countries.map(country => <Option value={country} key={country} > { country } </Option> ) }
            </Select>
            <Input pattern="[0-9]*" name="zip" type="text" />

            <ThemeButton theme="shop" >
                <Link to={routes.checkoutPayment}>
                    Go to payment
                </Link>
            </ThemeButton>  
        </form>
    </div>
}