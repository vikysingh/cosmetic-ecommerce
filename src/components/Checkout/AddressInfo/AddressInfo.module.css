.CHECKOUT__LEFT_COL {
    width: 100%;
    height: auto;
    padding: 5% 0 5% 0;
}

.CHECKOUT__LEFT_COL h3, .CHECKOUT__LEFT_COL h4 {
    font-weight: lighter;
    margin-top: 20px;
    margin-bottom: 5px;
}

.CHECKOUT__LEFT_COL input {
    height: 40px;
    margin: 5px 0;
    border-radius: 3px;
    border: 1px solid rgb(146, 146, 146);
}

.CHECKOUT__LEFT_COL form {
    margin-top: 30px;
    width: 80%;
}

.CHECKOUT__LEFT_COL > form > input:nth-child(2) {
    width: 100%;
}

.CHECKOUT__LEFT_COL form > input:nth-child(4),
.CHECKOUT__LEFT_COL form > input:nth-child(5),
.CHECKOUT__LEFT_COL form > input:nth-child(10),
.CHECKOUT__LEFT_COL__select {
    width: calc(50% - 5px) !important;
}

.CHECKOUT__LEFT_COL form > input:nth-child(4), .CHECKOUT__LEFT_COL__select {
    margin-right: 10px;
}

.CHECKOUT__LEFT_COL__select {
    background-color: #fff;
    padding: 2px;
    border: 2px solid rgb(146, 146, 146) !important;
}

.CHECKOUT__LEFT_COL__select > div {
    border: none !important;
}

.CHECKOUT__LEFT_COL button {
    height: 50px;
    width: 160px;
    border: none;
    margin-top: 10px;
}

@media (max-width: 900px) {
    .CHECKOUT__LEFT_COL form {
        width: 95%;
    }
}