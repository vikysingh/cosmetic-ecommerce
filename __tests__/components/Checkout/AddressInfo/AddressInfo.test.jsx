import React from "react"
import { shallow } from "enzyme"

import AddressInfo from "../../../../components/Checkout/AddressInfo/AddressInfo"

test('Should render AddressInfo from Checkout', () => {
    const wrapper = shallow(<AddressInfo />)

    expect(wrapper).toMatchSnapshot()
})