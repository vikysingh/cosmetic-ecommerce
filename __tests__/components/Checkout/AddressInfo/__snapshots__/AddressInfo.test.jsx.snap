// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AddressInfo from Checkout 1`] = `
<div
  className="CHECKOUT__LEFT_COL"
>
  <Breadcrumb
    className="CHECKOUT__LEFT_COL__breadcrumb"
  >
    <BreadcrumbItem>
      <[object Object]
        to="/"
      >
         Home 
      </[object Object]>
    </BreadcrumbItem>
    <BreadcrumbItem>
      <[object Object]
        to="/cart"
      >
         Cart 
      </[object Object]>
    </BreadcrumbItem>
    <BreadcrumbItem>
      Address
    </BreadcrumbItem>
  </Breadcrumb>
  <Title
    level={3}
  >
     Contact information 
  </Title>
  <Input
    placeholder="Email"
    type="email"
  />
  <form>
    <Title
      level={4}
    >
       Shipping address 
    </Title>
    <Input
      placeholder="First Name"
      type="text"
    />
    <Input
      placeholder="Last Name"
      type="text"
    />
    <Input
      placeholder="Address"
      type="text"
    />
    <Input
      placeholder="Apartment, suite etc.(optional)"
      type="text"
    />
    <Input
      placeholder="City"
      type="text"
    />
    <Select
      bordered={true}
      choiceTransitionName=""
      className="CHECKOUT__LEFT_COL__select"
      filterOption={[Function]}
      optionFilterProp="children"
      placeholder="Country"
      showSearch={true}
      transitionName="slide-up"
    >
      <Option
        key="Italy"
        value="Italy"
      >
         
        Italy
         
      </Option>
      <Option
        key="France"
        value="France"
      >
         
        France
         
      </Option>
      <Option
        key="Germany"
        value="Germany"
      >
         
        Germany
         
      </Option>
      <Option
        key="United Kingdom"
        value="United Kingdom"
      >
         
        United Kingdom
         
      </Option>
      <Option
        key="Spain"
        value="Spain"
      >
         
        Spain
         
      </Option>
      <Option
        key="Portugal"
        value="Portugal"
      >
         
        Portugal
         
      </Option>
      <Option
        key="Canada"
        value="Canada"
      >
         
        Canada
         
      </Option>
      <Option
        key="United States"
        value="United States"
      >
         
        United States
         
      </Option>
    </Select>
    <Input
      name="zip"
      pattern="[0-9]*"
      type="text"
    />
    <[object Object]
      theme="shop"
    >
      <[object Object]
        to="/checkout/payment"
      >
        Go to payment
      </[object Object]>
    </[object Object]>
  </form>
</div>
`;
